themes/theme-gmd/pages/ProductGallery/components/Content/components/ImagesSlider/index.jsx

```diff
@@ -13,17 +13,28 @@ import connect from './connector';
  * @return {JSX}
  */
 const ProductGalleryImages = ({ initialSlide, images }) => {
-  if (!Array.isArray(images) || images.length === 0) {
-    return <div className={styles.container} />;
-  }
-
   const { zoom = {} } = useWidgetSettings('@shopgate/engage/product/Gallery') || {};
 
-  const { GalleryImage: galleryResolutions } = getProductImageSettings();
+  const fullImages = useMemo(() => {
+    if (!images || !images.length) {
+      return null;
+    }
+    const { GalleryImage: galleryResolutions } = getProductImageSettings();
+    const lowerResolution = galleryResolutions[0];
+    const biggerResolution = galleryResolutions[galleryResolutions.length - 1];
+    return {
+      low: images.map(src => getFullImageSource(src, lowerResolution)),
+      big: images.map(src => getFullImageSource(src, biggerResolution)),
+    };
+  }, [images]);
 
-  const resolution = galleryResolutions[galleryResolutions.length - 1];
+  const loaded = useLoadImage(fullImages && fullImages.big && fullImages.big[0]);
+
+  if (!Array.isArray(images) || images.length === 0) {
+    return <div className={styles.container} />;
+  }
 
-  const imagesWithResolutions = images.map(src => getFullImageSource(src, resolution));
+  const showImages = loaded ? fullImages.big : fullImages.low;
```

Comment:
```suggestion
  const imagesToShow = loaded ? fullImages.big : fullImages.low;
```
showImages sounds like it would be bool